docs/tools/terraform/reference/cookbook/m3db-m3agg-recipe.rst

```diff
@@ -0,0 +1,108 @@
+While you can do aggregation without an M3 Aggregator node, a dedicated metrics aggregator can help reduce the volume of time series data stored by using variable data point resolutions. This is especially true if you want the ability to automatically roll-up data over time.
+For example, to aggregate all IoT metrics from the last two months into 10-minute points. 
+
+Let's cook!
+-----------
+
+The following Terraform recipe will create an Aiven for M3 service, an Aiven for M3 Aggregator service, and the related service integration.
+
+.. Tip::
+
+    Be sure to check out the :doc:`getting started guide <../../get-started>` to learn about the common files required to execute the following recipe. For example, you'll need to declare the variables for ``project_name``, ``api_token``, and ``service_name_prefix``.
+
+``services.tf`` file:
+
+.. code:: terraform
+ 
+  resource "aiven_m3db" "demo-m3db" {
+    project      = var.project_name
+    cloud_name   = "google-northamerica-northeast1"
+    plan         = "business-8"
+    service_name = join("-", [var.service_name_prefix, "m3db"])
+
+    m3db_user_config {
+      m3db_version = 1.5
+
+      namespaces {
+        name = "m3_default_unaggregated_ns"
+        type = "unaggregated"
+      }
+      namespaces {
+        name       = "m3_lowRes_aggregated_ns"
+        type       = "aggregated"
+        resolution = "10m"
+        options {
+          retention_options {
+            retention_period_duration = "1d"
+          }
+        }
+      }
+      namespaces {
+        name       = "m3_medRes_ns"
+        type       = "aggregated"
+        resolution = "2m"
+        options {
+          retention_options {
+            retention_period_duration = "6h"
+          }
+        }
+      }
+      namespaces {
+        name       = "m3_highRes_ns"
+        type       = "aggregated"
+        resolution = "10s"
+        options {
+          retention_options {
+            retention_period_duration = "1h"
+          }
+        }
+      }
+    }
+  }
+
+  resource "aiven_m3aggregator" "demo-m3a" {
+    project      = var.project_name
+    cloud_name   = "google-northamerica-northeast1"
+    plan         = "business-8"
+    service_name = join("-", [var.service_name_prefix, "m3a"])
+
+    m3aggregator_user_config {
+      m3aggregator_version = 1.5
+    }
+  }
+
+  resource "aiven_service_integration" "int-m3db-aggr" {
+    project                  = var.project_name
+    integration_type         = "m3aggregator"
+    source_service_name      = aiven_m3db.demo-m3db.service_name
+    destination_service_name = aiven_m3aggregator.demo-m3a.service_name
+  }
+
+``namespaces`` in M3 is used to determine how metrics are stored and retained. There is always one unaggregated namespace which is configured under the ``demo-m3db`` resource ``namespaces`` block. There are three aggregated namespaces defined within the same block for different resolution settings.
```

Comment:
When you mention the 3 aggregated namespaces it would be good to reference the names and what each one does like
* `m3_highres_ns` downsamples the data to 10s and keeps the data (retention time) for 1h
etc

Reply:
Good call. Sending that improvement via next commit.